docs/guides/socket-collector.md

```diff
@@ -0,0 +1,74 @@
+---
+title: 'The "socket-collector" gadget'
+weight: 10
+---
+
+socket-collector gathers information about TCP and UDP sockets.
+
+We will start this demo by using nginx to create a web server on port 80:
+```
+$ kubectl create namespace test-socketcollector
+namespace/test-socketcollector created
+$ kubectl run --restart=Never --namespace test-socketcollector --image=nginx nginx-app --port=80
+pod/nginx-app created
+```
+
+Wait for the pod to get ready:
+```
+$ kubectl wait --timeout=-1s --namespace test-socketcollector --for=condition=ready pod/nginx-app ; kubectl get pod --namespace test-socketcollector
+pod/nginx-app condition met
+NAME        READY   STATUS    RESTARTS   AGE
+nginx-app   1/1     Running   0          46s
+```
+
+We will now use Inspektor Gadget to retrieve the TCP/UDP sockets information
+of the nginx-app pod. Notice we are filtering by namespace but we could have
+done it also using the podname or labels:
+```
+$ kubectl gadget socket-collector --namespace test-socketcollector
+NODE       NAMESPACE               POD          PROTOCOL    LOCAL         REMOTE       STATUS
+my-node    test-socketcollector    nginx-app    TCP         0.0.0.0:80    0.0.0.0:0    LISTEN
+```
+
+In the output, "LOCAL" is the local IP address and port number pair.
+If connected, "REMOTE" is the remote IP address and port number pair,
+otherwise, it will be "0.0.0.0:0". While "STATUS" is the internal
+status of the socket.
+
+Now, modify the nginx configuration to listen on port 8080 instead of 80 and reload the daemon:
+```
+$ kubectl exec --namespace test-socketcollector nginx-app -- /bin/bash -c "sed -i 's/listen \+80;/listen\t8080;/g' /etc/nginx/conf.d/default.conf && exec nginx -s reload"
+[...] signal process started
+```
```

Comment:
Perhaps you can start a second nginx instance with a different port to avoid changing the confirmation for this one?

Reply:
Does it look that bad 😅? My idea was actually to show that the information got updated (The entry for 80 was removed and a new one for 8080 was added).

Reply:
Well, it's not bad, just that I don't want readers get distracted by that long command to change the port.

Reply:
Understood. I would merge this PR as it is for now and, as discussed also with Alban, I will add more tests to show TCP sockets in `Established` status and also UDP sockets. At that point, we can decide whether to keep this test. WDYT?

Reply:
SGTM!